src/test/services/errorClear.test.tsx

```diff
@@ -0,0 +1,74 @@
+import { screen, within } from "@testing-library/react";
+import { fillTextField, renderTestEnvironment } from "../utils";
+import userEvent from "@testing-library/user-event";
+import { rest } from "msw";
+import { API_PATH } from "../../app/services";
+import * as actions from "../../app/state/actions";
+
+describe("ErrorClear", () => {
+  it("If a error is received from API and stored into state, this will display on the page, but when changing route it will clear", async () => {
+    const clearErrorSpy = jest.spyOn(actions, 'clearError');
+
+    renderTestEnvironment({
+      role: "ANONYMOUS",
+      extraEndpoints: [
+        rest.post(API_PATH + "/auth/SEGUE/authenticate", (req, res, ctx) => {
+          return res(
+            ctx.status(401),
+            ctx.json({
+              errorMessage: "Incorrect credentials provided.",
+              bypassGenericSiteErrorPage: false,
+            })
+          );
+        }),
+      ],
+    });
+    // Locate "Log In" in the header and click on it
+    const header = await screen.findByTestId("header");
+    const logIn = within(header).getByRole("link", { name: "LOG IN" });
+    await userEvent.click(logIn);
+    // Wait for Log In page to load and check that the page title is correct
+    const title = screen.getByRole("heading", {
+      name: /log in or sign up:/i,
+    });
+    expect(title).toBeInTheDocument();
+    // Fill in the form with incorrect details to trigger an error
+    const emailInput = screen.getByRole("textbox", {
+      name: /email address/i,
+    });
+    const passwordInput = screen.getByLabelText(/password/i);
+    await fillTextField(emailInput, "randomTestEmail@test.com");
+    await fillTextField(passwordInput, "randomTestPassword123!");
+    // Click on the Log In button
```

Comment:
I am quite allergic to comments: most of the time, you can exchange the comment and subsequent lines for a function. This will make your code tidier, more reusable, and it will make this function shorter and easier to read. One of the problems with comments is that they often lie: they don't tell the whole truth or, with the pass of time, someone updates the code underneath and forgets to update the comment, so they're out of sync and the comment becomes not just useless, but dangerous (because it's confusing, as it's incorrect).

For instance, this can be extracted into a function called `clickOnTheLogInButton`. The code above this one can be extracted into a function called `fillInTheFormWithIncorrectDetails`. And the same goes for most of the code in this file.

What do you think? Does this make sense?

Reply:
I actually love comments in code 😀 however I appreciate the point about the maintenance overhead and potential for issues.  I've removed some of the comments and added in a `fillFormIncorrectly` specifically for that test, as well as a utility for `clickButton` which is a little more generic and re-usable in other tests.